### PR TITLE
Week 2 assessment jerrod2023

### PR DESCRIPTION
**Pull request by jerrod2023**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Career Readiness
- [X] Coding Challenges
- [X] Coding Critique
- [X] Coding Concepts
- [X] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: career-readiness.md [hunk 1]
```diff
@@ -8,17 +8,17 @@ This week, we covered the importance of a great technical portfolio to help you
...
 
-- [ ] add additional previous jobs (up to 10 years as appropriate) in the Experience section. Add skills and descriptions to your jobs that highlight transferable skills.
-- [ ] create a new post about your first two weeks at LEARN. Topics can include (but are not limited to) JavaScript, git and GitHub, pair programming, or communication skills.
-- [ ] add at least ten more professional connections. Connections can include your cohort mates, LEARN alumni, LEARN staff, following companies of interest, friends and family, and former coworkers.
-- [ ] react (like, celebrate, comment, etc) to ten posts from your existing connections.
+- [X] add additional previous jobs (up to 10 years as appropriate) in the Experience section. Add skills and descriptions to your jobs that highlight transferable skills.
+- [X] create a new post about your first two weeks at LEARN. Topics can include (but are not limited to) JavaScript, git and GitHub, pair programming, or communication skills.
+- [X] add at least ten more professional connections. Connections can include your cohort mates, LEARN alumni, LEARN staff, following companies of interest, friends and family, and former coworkers.
```
Comment (chrisyhlee) [inline career-readiness.md:23]: Keep connecting with people!

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,39 +13,97 @@
...
-const numbersArray1 = [6, 7, 8, 9, 10]
-// Expected output: [18, 21, 24, 27, 30]
-const numbersArray2 = [24, 27, 30, 33, 36]
-// Expected output: [72, 81, 90, 99, 108]
+// describe("numMultBy3", () => {
+//     it("takes in an array of numbers and returns an array with all the numbers multiplied by 3.", () => {
+//         const numbersArray1 = [6, 7, 8, 9, 10]
+//         const numbersArray2 = [24, 27, 30, 33, 36]
+//       //expect(numMultBy3r(numbersArray1)).toEqual([18, 21, 24, 27, 30])
+//       //expect(numMultBy3r(numbersArray1)).toEqual([72, 81, 90, 99, 108])
```
Comment (chrisyhlee) [inline code-challenges.test.js]: You should have different arguments in each expect. The way this is set up, you have 2 different expected outcomes for the same argument, which doesn't make sense.

File: code-challenges.test.js [hunk 3]
```diff
@@ -13,39 +13,97 @@
...
+// const numbersArray1 = [6, 7, 8, 9, 10]
+// const numbersArray2 = [24, 27, 30, 33, 36]
+// // Pseudo code:
+//     //Create a Function = "numMultBy3"
+//     ////.map method to return the new arrays    that will multiply by 3
+//     // //input: Arrays of [6, 7, 8, 9, 10] and [24, 27, 30, 33, 36]
+//     //output:
+// const numMultBy3 = (array) => { 
+//     return array.map(numbersArray1 => numbersArray1 * 3) 
+//     return array.map(numbersArray2 => numbersArray2 * 3)
```
Comment (chrisyhlee) [inline code-challenges.test.js]: A function should only have 1 return

File: code-challenges.test.js [hunk 4]
```diff
@@ -13,39 +13,97 @@
...
-const object1 = { number: 15 }
-// Expected output: "15 is divisible by three"
-const object2 = { number: 0 }
-// Expected output: "0 is divisible by three"
-const object3 = { number: -7 }
+// describe("numDivBy3", (number) => {
+//     it("takes a object as an argument and decides if the number inside it is evenly divisible by three or not.", () => {
+//       expect(numDivBy3()).toEqual("15 is divisible by three")
+//       expect(numDivBy3()).toEqual("0 is divisible by three")
+//       expect(numDivBy3()).toEqual("-7 is divisible by three")
```
Comment (chrisyhlee) [inline code-challenges.test.js:63]: You have no arguments in these function calls

File: coding-concepts.js [hunk 5]
```diff
@@ -9,28 +9,31 @@
...
+  return `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+// the function body contains a template string and not returned explicity
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: error
+// b) Verify and explain: greeter function = takes 1 parameter (name)
```
Comment (chrisyhlee) [inline coding-concepts.js]: undefined is not the same as an error

**Review by chrisyhlee — APPROVED** [review]
I'm seeing difficulty understanding the fundamentals of JavaScript. Your main focus should be taking time outside of class to study, review, and study some more! It's essential to get these fundamentals down, or you won't be able to build on top of it as you move forward in the cohort. Figure out your study preferences and put that work in! You got this, just need to keep practicing. This assessment displays Novice level. If you are interested in extra 1 on 1 tutoring, reach out to myself, Elyse or Gene.

**Comment by jerrod2023** [comment]
Yes, I am a bit struggling understanding all of this. I’m still am trying and giving some more hours to this on the weekends too.Sent from my iPhoneOn Oct 16, 2023, at 14:03, Chris ***@***.***> wrote:﻿
@chrisyhlee approved this pull request.

I'm seeing difficulty understanding the fundamentals of JavaScript. Your main focus should be taking time outside of class to study, review, and study some more! It's essential to get these fundamentals down, or you won't be able to build on top of it as you move forward in the cohort. Figure out your study preferences and put that work in! You got this, just need to keep practicing. This assessment displays Novice level. If you are interested in extra 1 on 1 tutoring, reach out to myself, Elyse or Gene.

In career-readiness.md:
>  
 Check the boxes below to certify that you have completed the following items on your LinkedIn profile:
 
-- [ ] add additional previous jobs (up to 10 years as appropriate) in the Experience section. Add skills and descriptions to your jobs that highlight transferable skills.
-- [ ] create a new post about your first two weeks at LEARN. Topics can include (but are not limited to) JavaScript, git and GitHub, pair programming, or communication skills.
-- [ ] add at least ten more professional connections. Connections can include your cohort mates, LEARN alumni, LEARN staff, following companies of interest, friends and family, and former coworkers.
-- [ ] react (like, celebrate, comment, etc) to ten posts from your existing connections.
+- [X] add additional previous jobs (up to 10 years as appropriate) in the Experience section. Add skills and descriptions to your jobs that highlight transferable skills.
+- [X] create a new post about your first two weeks at LEARN. Topics can include (but are not limited to) JavaScript, git and GitHub, pair programming, or communication skills.
+- [X] add at least ten more professional connections. Connections can include your cohort mates, LEARN alumni, LEARN staff, following companies of interest, friends and family, and former coworkers.

Keep connecting with people!



In code-challenges.test.js:
> @@ -13,39 +13,97 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const numbersArray1 = [6, 7, 8, 9, 10]
-// Expected output: [18, 21, 24, 27, 30]
-const numbersArray2 = [24, 27, 30, 33, 36]
-// Expected output: [72, 81, 90, 99, 108]
+// describe("numMultBy3", () => {
+//     it("takes in an array of numbers and returns an array with all the numbers multiplied by 3.", () => {
+//         const numbersArray1 = [6, 7, 8, 9, 10]
+//         const numbersArray2 = [24, 27, 30, 33, 36]
+//       //expect(numMultBy3r(numbersArray1)).toEqual([18, 21, 24, 27, 30])
+//       //expect(numMultBy3r(numbersArray1)).toEqual([72, 81, 90, 99, 108])

You should have different arguments in each expect. The way this is set up, you have 2 different expected outcomes for the same argument, which doesn't make sense.



In code-challenges.test.js:
> +// }
+
+  
+// // b) Create the function that makes the test pass.
+
+
+// const numbersArray1 = [6, 7, 8, 9, 10]
+// const numbersArray2 = [24, 27, 30, 33, 36]
+// // Pseudo code:
+//     //Create a Function = "numMultBy3"
+//     ////.map method to return the new arrays    that will multiply by 3
+//     // //input: Arrays of [6, 7, 8, 9, 10] and [24, 27, 30, 33, 36]
+//     //output:
+// const numMultBy3 = (array) => { 
+//     return array.map(numbersArray1 => numbersArray1 * 3) 
+//     return array.map(numbersArray2 => numbersArray2 * 3)

A function should only have 1 return



In code-challenges.test.js:
>  
 // --------------------1) Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const object1 = { number: 15 }
-// Expected output: "15 is divisible by three"
-const object2 = { number: 0 }
-// Expected output: "0 is divisible by three"
-const object3 = { number: -7 }
+// describe("numDivBy3", (number) => {
+//     it("takes a object as an argument and decides if the number inside it is evenly divisible by three or not.", () => {
+//       expect(numDivBy3()).toEqual("15 is divisible by three")
+//       expect(numDivBy3()).toEqual("0 is divisible by three")
+//       expect(numDivBy3()).toEqual("-7 is divisible by three")

You have no arguments in these function calls



In coding-concepts.js:
>  
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: error
+// b) Verify and explain: greeter function = takes 1 parameter (name)

undefined is not the same as an error

**Comment by jerrod2023** [comment]
added changes to my coding-concepts, code-challenges, [X] add additional previous jobs, [X] create a new post about your first two weeks at LEARN, [X] add at least ten more professional connections, and [X] react to 10 post.

**Pull request merged**